packages/site/cms-site/src/cookies/CookieSafe.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+import React from "react";
+
+import { usePreview } from "../preview/usePreview";
+
+type Props = React.PropsWithChildren<{
+    consented: boolean;
+    fallback: React.ReactNode;
+}>;
+
+// TODO: The Fallback is currently also rendered, when the page has loaded, but the cookie-banner has not been loaded yet.
+// - Maybe store the consented cookies in local-storage and use that value until the cookie-banner has been loaded.
```

Comment:
@johnnyomair the issue here is that the fallback is currently shown as soon as the page loads, then the cookie-provider takes some time to load and only then, if the cookies have been consented to, the fallback is replaced by the actual content. 

This isn't optimal, as the content flickers and is replaced, about a second after the page initially loads. 

Do you think my `localStorage` suggestion is a good idea? 
Do you have any other idea how we could fix this issue? 
Maybe on the server side, using cookies 😅

Reply:
> Do you think my localStorage suggestion is a good idea?

I don't think this is a good idea since it would in theory "survive" a clear cookies command.

> Do you have any other idea how we could fix this issue?

Maybe your API should have some sort of "pending" or "undefined" state, where neither the content nor the fallback is shown?

> Maybe on the server side, using cookies 😅

I doubt this is possible, especially since we usually have a CDN for the site.

Reply:
> Maybe your API should have some sort of "pending" or "undefined" state, where neither the content nor the fallback is shown?

Yes, that's probably the way.

Reply:
I added a loading state here: [Add placeholder for state where cookie consent is unknown](https://github.com/vivid-planet/comet/pull/2193/commits/2dc7534ac4ed6d9fa9d48d762214ef395d23738c)